modules/shared/k8shelpers/transport_test.go

```diff
@@ -0,0 +1,36 @@
+package k8shelpers
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var TEST_TOKEN = "eyJhbGciOiJSUzI1NiIsImtpZCI6IjEyMzQ1NiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwic3ViIjoic3lzdGVtOnNvbWUtbmFtZXNwYWNlOnRlc3QtdXNlciIsImF1ZCI6Imt1YmVybmV0ZXMiLCJleHAiOjQ3Mjg5MjQ4MDAsImlhdCI6MTY1MDkzMjAwMH0.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+
+func TestBearerTokenRoundTripper_headerSet(t *testing.T) {
+	testserver := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		header, ok := r.Header["Authorization"]
```

Comment:
The test doesn't wait for this function to be called before exiting. To ensure that the function gets executed during the test you can use a `sync.WaitGroup`.